code/controllers/subsystem/objects.dm

```diff
@@ -17,14 +17,12 @@ var/datum/subsystem/objects/SSobj
 
 /datum/subsystem/objects/Initialize(timeofday, zlevel)
 	setupGenetics()
-	for(var/atom/movable/AM in world)
-		if (zlevel && AM.z != zlevel)
+	for(var/atom/A in world)
+		if (zlevel && A.z != zlevel)
```

Comment:
It will shortcircuit anyway ? Or i'm missing what you're saying here. Also making this init,init_level,init_block with per atom helper or similar. Maybe later.